android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java

```diff
@@ -169,6 +170,22 @@ public RCTMGLMapView(Context context, RCTMGLMapViewManager manager, MapboxMapOpt
         mHandler = new Handler();
 
         setLifecycleListeners();
+            
+        addOnAttachStateChangeListener(new OnAttachStateChangeListener() {
+            @Override
+            public void onViewAttachedToWindow(View v) {
+                if(mLocationLayer != null){
+                    mLocationLayer.onStart();
```

Comment:
@nitaliano might want to wrap this onStart call with the same logic in [setReactShowUserLocation()](https://github.com/mapbox/react-native-mapbox-gl/blob/master/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java#L700)

```setReactShowUserLocation(mShowUserLocation)```